addons/sale_stock/views/sale_order_views.xml

```diff
@@ -57,6 +57,9 @@
             <field name="model">sale.order</field>
             <field name="inherit_id" ref="sale.sale_order_tree"/>
             <field name="arch" type="xml">
+                <xpath expr="//field[@name='commitment_date']" position="attributes">
+                    <attribute name="decoration-danger">((commitment_date &lt; effective_date or commitment_date &lt; datetime.datetime.combine(datetime.date.today(), datetime.time(0,0,0)).to_utc().strftime('%Y-%m-%d %H:%M:%S')) and delivery_status == 'pending' and effective_date &lt;= commitment_date or delivery_status == 'partial')</attribute>
+                </xpath>
```

Comment:
I think this `xpath` should be in a `view_order_tree` tree view not in this tree view 
Probably because `delivery_status` only set when SO confirmed and not in Quotation :thinking: so there is no point of doing this xpath here :thought_balloon: 

Can you check?

Also try to split line into `< 100` charactors :innocent:

Reply:
Done the changes :+1: 
Thankyou so much